### PR TITLE
nixification

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -3,14 +3,29 @@
...
     ];
 
+    preConfigure = ''
+      cp -r ${npmToNix { inherit src; }} node_modules
+      chmod -R +w node_modules
+      patchShebangs node_modules
+    '';
+
+    buildPhase = ''
+      make docs
```
Comment (peeech) [inline default.nix]: `make docs` builds chaperone docs, not chaperone

Comment (zo-el) [reply default.nix]: @peeech Sorry, I didn't push up my last commit. 
I will do it now

File: default.nix [hunk 2]
```diff
@@ -3,14 +3,37 @@
 with pkgs;
 
 {
-  chaperone = stdenv.mkDerivation {
+  chaperone = stdenv.mkDerivation rec {
     name = "chaperone";
-    src = gitignoreSource ./.;
+    src = ./.;
```
Comment (yegortimoshenko) [inline default.nix]:
```suggestion
    src = gitignoreSource ./.;
```

File: default.nix [hunk 3]
```diff
@@ -3,14 +3,37 @@
...
 
     nativeBuildInputs = with buildPackages; [
       nodejs-12_x
     ];
 
-    doCheck = false;
+    comb = ./comb;
```
Comment (yegortimoshenko) [inline default.nix]:
```suggestion
```

File: default.nix [hunk 4]
```diff
@@ -3,14 +3,37 @@
...
+
+    preConfigure = ''
+      cp -r ${npmToNix { inherit src; }} node_modules
+      chmod -R +w node_modules
+      chmod +x node_modules/.bin/webpack
+      patchShebangs node_modules
+      cp -r ${npmToNix { src = comb; }} comb/node_modules
```
Comment (yegortimoshenko) [inline default.nix]:
```suggestion
      cp -r ${npmToNix { src = "${./comb}"; }} comb/node_modules
```

**Comment by yegortimoshenko** [comment]
Also commit message is meh :)

File: default.nix (hunk 2 above)
Comment (zo-el) [reply default.nix]: @yegortimoshenko We cannot use this because there are some gitignored files are needed for the build.

Comment (yegortimoshenko) [reply default.nix]: Then that should be fixed prior to merge. This will break Holo Nixpkgs if you leave it as-is.

**Pull request closed without merging**